aws-push-notifications-pinpoint-common/src/main/java/com/amplifyframework/pushnotifications/pinpoint/PushNotificationsUtils.kt

```diff
@@ -106,12 +106,12 @@ class PushNotificationsUtils(
     @SuppressLint("NewApi")
     fun showNotification(
         notificationId: Int,
-        payload: NotificationPayload,
+        payload: PinpointNotificationPayload,
         targetClass: Class<*>?
     ) {
         CoroutineScope(Dispatchers.IO).launch {
             val largeImageIcon = payload.imageUrl?.let { downloadImage(it) }
-            val notificationIntent = Intent(context, targetClass)
+            val notificationIntent = Intent(context, payload.targetClass ?: targetClass)
```

Comment:
Rather than doing this here, could we instead add else on the target class when doing `PinpointNotificationPayload.fromNotificationPayload(notificationPayload)`?

Reply:
As per discussion, this is not a problem.